extensions/wikia/AppPromoLanding/AppPromoLandingController.class.php

```diff
@@ -139,16 +148,16 @@ public function content( $params ){
 						//'url' => $item['url'],
 						'title' => $item['title'],
 						'imgUrl' => $thumbnail,
-						'width' => self::IMG_WIDTH,
-						'height' => self::IMG_HEIGHT
+						'width' => static::IMG_WIDTH,
+						'height' => static::IMG_HEIGHT
 					];
 				}
 			}
 		}
 
 		// Not all articles will have images, so we may have more or less than we need. Here,
 		// we will right-size the array.
-		$numThumbsNeeded = ( self::THUMBS_NUM_ROWS * self::THUMBS_PER_ROW );
+		$numThumbsNeeded = ( static::THUMBS_NUM_ROWS * static::THUMBS_PER_ROW );
 		if(count( $trendingArticles ) > $numThumbsNeeded){
 			$trendingArticles = array_slice($trendingArticles, 0, $numThumbsNeeded);
 		} else if(count( $trendingArticles ) < $numThumbsNeeded ){
```

Comment:
Since we don't need any help shuffling (afaik) and array_rand gives a warning when trying to pick more than exist in the source-array, I don't think it would help us in this particular instance.